src/core/playground/authenticationKey.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { sha3_256 as sha3Hash } from "@noble/hashes/sha3";
+import { HexInput, SigningScheme as AuthenticationKeyScheme } from "../../types";
+
+import { AccountAddress } from "../accountAddress";
+import { Hex } from "../hex";
+
+/**
+ * Each account stores an authentication key. Authentication key enables account owners to rotate
+ * their private key(s) associated with the account without changing the address that hosts their account.
+ * @see {@link https://aptos.dev/concepts/accounts | Account Basics}
+ *
+ * Note: AuthenticationKey only supports Ed25519 and MultiEd25519 public keys for now.
+ *
+ * Account addresses can be derived from AuthenticationKey
+ */
+export class AuthenticationKey {
+  /**
+   * An authentication key is always a SHA3-256 hash of data, and is always 32 bytes.
+   */
+  static readonly LENGTH: number = 32;
+
+  /**
+   * The raw bytes of the authentication key.
+   */
+  public readonly data: Hex;
+
+  constructor(args: { data: HexInput }) {
+    const { data } = args;
+    const hex = Hex.fromHexInput(data);
+    if (hex.toUint8Array().length !== AuthenticationKey.LENGTH) {
+      throw new Error(`Authentication Key length should be ${AuthenticationKey.LENGTH}`);
+    }
+    this.data = hex;
+  }
+
+  toString(): string {
+    return this.data.toString();
+  }
+
+  toUint8Array(): Uint8Array {
+    return this.data.toUint8Array();
+  }
+
+  static fromSchemeAndBytes({
+    scheme,
+    input,
+  }: {
+    scheme: AuthenticationKeyScheme;
+    input: HexInput;
+  }): AuthenticationKey {
+    const inputBytes = Hex.fromHexInput(input).toUint8Array();
```

Comment:
I couldn't find it in this PR, but we have auth key generation based on scheme logic
https://github.com/aptos-labs/aptos-ts-sdk/blob/main/src/core/authenticationKey.ts#L79

Reply:
Yes, what I'm proposing here is to have each public key define how to derive its authentication key, as opposed to have a switch somewhere else that needs to handle all variants of public keys and schemas.

Reply:
ohh is the public key handling on this PR? or plan to add it later?

Reply:
It's in this PR already

for AnyPublicKey
https://github.com/aptos-labs/aptos-ts-sdk/blob/8642418bd1b6b1ea947afc54377183cef2ca808e/src/core/playground/wrapped.ts#L50-L55

for Ed25519PublicKey
https://github.com/aptos-labs/aptos-ts-sdk/blob/8642418bd1b6b1ea947afc54377183cef2ca808e/src/core/playground/ed25519.ts#L47-L52

for LegacyEd25519PublicKey
https://github.com/aptos-labs/aptos-ts-sdk/blob/8642418bd1b6b1ea947afc54377183cef2ca808e/src/core/playground/legacy/multiEd25519.ts#L128-L133